File: exercises/dotnet/McrDigital.Bootcamp1.Cards/AnimalCard.cs
namespace McrDigital.Bootcamp1.Cards
{
    public class AnimalCard : ICard
    {
        private readonly Animal _animal;

        public AnimalCard(Animal animal)
        {
            _animal = animal;
        }

        public Animal Animal
        {
            get => _animal;
        }

        public bool Snap(ICard other)
        {
            return _animal == ((AnimalCard) other)._animal;
        }

        public override string ToString()
        {
            return _animal.ToString();
        }
    }
}

File: exercises/php/cards/AnimalCard.php
<?php

namespace McrDigital\PhpFundamentals1\Cards;

class AnimalCard
{
    private string $animal;

    public function __construct(string $animal)
    {
        $this->animal = $animal;
    }

    public function __toString(): string
    {
        return $this->animal;
    }

    public function snap(AnimalCard|null $card): bool
    {
        return !!$card && $this->animal === $card->animal;
    }

}